coprocess_python.go

```diff
@@ -86,18 +84,20 @@ static void Python_HandleMiddlewareCache(char* bundle_path) {
 }
 
 static int Python_NewDispatcher(char* middleware_path, char* event_handler_path, char* bundle_paths) {
+	PyThreadState*  mainThreadState = PyEval_SaveThread();
+	gilState = PyGILState_Ensure();
```

Comment:
Why do we need both `runtime.LockOSThread` and this? Is this an improvement that's unrelated to the fix?